### PR TITLE
feat(ui): rating section component

### LINKED ISSUE #45: [Feat][UI] rating section component
## Mockups / Screenshots

ReviewRatingSection Component

image

https://www.figma.com/file/nSCkMgGjpXTgLFEHykk0yb/Product-Design?type=design&node-id=324-21020&mode=dev

StatItem component

image

https://www.figma.com/file/nSCkMgGjpXTgLFEHykk0yb/Product-Design?type=design&node-id=324%3A22552&mode=dev

## Implementation

ReviewRatingSection component 
- `/common/components/ReviewRatingSection`
- Create `ReviewRatingSection.theme.ts` to manage variants and size. Refer to #19 for guide
- Use the other components that are already available as much as possible

ReviewRatingSection props 
- `isVisible`: `bool` - controls if section is visible or not (not visible section will have the `"Login to see rating"` text)
- `ratingItems`: `{ label: string, rating: number}` 
  where `label` refers to `ReviewLabelName` (like "Engaging", "Managable Workload" etc)
  and `rating` refers to the percentage representation of the statistic (like `0.70` to mean 70%, `0.105` to mean 10.5% etc)
- See figma for other props like `size`. `state` should not be a prop, but rather using pseudo classes
- Spread the rest of the props into `<ReviewRatingSection {...props}>`

StatItem component 
- `/common/components/StatItem`
- Create `StatItem.theme.ts` to manage variants and size
- Use the other components that are already available as much as possible

### PR DESCRIPTION
**Pull request by DavidLHW**
closes #45 

## Changes

- add `RatingSection` component
- add `StatItem` component
- add `LockCtaOverlay` component
- add `formatPercentage` function
- add `HeartIcon`
- fix `Dialog` component not passing props to dialog primitive
- add usage examples at [`/components`]()

## Preview

| | |
|---|---|
| <div style="width:290px">Default</div> |   |
| Default<br>(light) |  |
| Locked |  |
| Locked<br>(light) |  |
| `StatItem`<br>(default) |  |
| `StatItem`<br>(default,light) |  |
| `StatItem`<br>(locked) |  |
| `StatItem`<br>(locked,light) |  |
| `StatItem`<br>(horizontal) |  |
| `StatItem`<br>(horizontal,light) |  |
| `StatItem`<br>(horizontal,locked) |  |
| `StatItem`<br>(horizontal,locked,light) |  |

### CODE REVIEW

File: src/common/components/RatingSection/RatingSection.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { formatPercentage } from "@/common/functions/format";
+
+export type RatingItem = {
+  label: string;
+  rating: number;
+};
+
+export type RatingSectionProps = RatingSectionVariants & {
+  headingRatingItem: RatingItem;
+  ratingItems: RatingItem[];
```
Comment (Aztriltus) [inline src/common/components/RatingSection/RatingSection.tsx:17]: Could we just use the props of StatItem? So if the API changes for that component, we would be prompted to update this component too

File: src/common/functions/format.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const formatPercentage = (amount: number) => {
```
Comment (Aztriltus) [inline src/common/functions/format.ts:1]: Do you think we should keep one function to one file? I think having a file for each makes it easier to glance what are the available functions. Wdyt?

File: src/common/components/RatingSection/RatingSection.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          <StatItem
+            label={headingRatingItem.label}
+            rating={headingRatingItem.rating}
+            layout="horizontal"
+            isLocked={isLocked}
+          />
+        </div>
+      </div>
+      <div className={statItemWrapper()}>
+        {ratingItems.map((item, i) => (
```
Comment (Aztriltus) [inline src/common/components/RatingSection/RatingSection.tsx:43]: Once we use StatItem[] then we can just spread it in, with the same overrides you have e.g isLocked

**Review by Aztriltus — CHANGES_REQUESTED** [review]
Some comments!

File: src/common/components/RatingSection/RatingSection.tsx (hunk 1 above)
Comment (DavidLHW) [reply src/common/components/RatingSection/RatingSection.tsx]: sure! updated at a0c73fe

File: src/common/functions/format.ts (hunk 2 above)
Comment (DavidLHW) [reply src/common/functions/format.ts]: sounds good, updated at 7aca4ea and added this to the functions readme at d4aaec1

File: src/common/components/RatingSection/RatingSection.tsx (hunk 3 above)
Comment (DavidLHW) [reply src/common/components/RatingSection/RatingSection.tsx:36]: updated at a0c73fe!

**Review by Aztriltus — APPROVED** [review]
Nice, thanks for the changes. LGTM

**Pull request merged**